pages/[[...slug]].js

```diff
@@ -31,26 +34,35 @@ export const getStaticPaths = async () => {
 };
 
 export const getStaticProps = async (context) => {
-    const {
-        params: { slug }
-    } = context;
     try {
-        let url = slug ? `/${slug.join('/')}` : '/index';
+        const {
+            params: { slug }
+        } = context;
+
+        const [languageIso] = slug || [];
+        const { languageId, hasLanguages, ...rest } = await getLanguagesProps(languageIso);
+
+        const url = await getPageUrl(slug, hasLanguages);
+
         // Fetch the page object from DotCMS Page API
-        const pageRender = await getPage(url);
+        const pageRender = await getPage(url, languageId);   
+        
         // Fetch the navigation from DotCMS Navigation API
         const nav = await getNav('4');
 
         return {
             props: {
                 pageRender,
-                nav
+                nav,
+                languageProps: { languageId, hasLanguages, ...rest }
             },
             revalidate: 1
         };
     } catch (error) {
         return {
-            props: { error }
+            props: {
+                error: isJson(error.message) ? JSON.parse(error.message) : error
```

Comment:
I think we are not sending the `statusCode` here and we use it here https://github.com/dotCMS/dotcms-spa/blob/091246d6445e6d6c43d665d5af651fe754be3b47/components/layout/ErrorPage.js#L35